app/src/test/java/org/mozilla/fenix/search/toolbar/ToolbarViewTest.kt

```diff
@@ -158,6 +160,161 @@ class ToolbarViewTest {
         verify { editToolbar.setIcon(any(), "Search Engine") }
     }
 
+    @Test
+    fun `GIVEN a general search engine is default WHEN a topic specific engine is selected THEN the hint changes`() {
+        val generalEngine = SearchEngine(
+            id = UUID.randomUUID().toString(),
+            name = "General",
+            icon = testContext.getDrawable(R.drawable.ic_search)!!.toBitmap(),
+            type = SearchEngine.Type.BUNDLED,
+            resultUrls = listOf(
+                "https://www.startpage.com/sp/search?q={searchTerms}",
+            ),
+            isGeneral = true,
+        )
+        val topicSpecificEngine = SearchEngine(
```

Comment:
Looks like the `type` and `isGeneral` properties is all that needs to be different between the search engines  used in this class.
Could we have a private function private to this file with the above properties received as arguments which would return a new search engine? Seems like it would help eliminate some duplicate code.